src/matching/match_basins.jl

```diff
@@ -0,0 +1,41 @@
+# TODO: I don't know how this fits into the new interface...
```

Comment:
Can you show me how it can be adapted to the new interface with a concrete example? In the current interface the inputs are dictionaries of IDs => values. Here we have an array whose entries are the IDs. Fundamentally different.

One idea I have for adapting to new interface is to transform the basin to dictionarys ID => vector of cartesian indices with this ID.

Reply:
Ok, tomorrow I will work on this one.

Reply:
@awage haha have a look at the latest commit I just pushed. I was working on this today and I ended up doing **exactly** the same thing you thought. :D

Reply:
This method is actually just as efficient as the previous method. Why do you think it is less efficient?

Reply:
Not efficient in terms of extra memory allocation. But this is not the computational bottleneck. Basins computation is.

Reply:
There is no extra memory allocation happening. The operation `findall(isequal(k), b₋) for k in unique(b₋)` is always done in the end of the day regardless of how we will represent the basins.